pilot/pkg/config/kube/ingressv1/controller.go

```diff
@@ -159,50 +165,37 @@ func (c *controller) shouldProcessIngress(mesh *meshconfig.MeshConfig, i *knetwo
 }
 
 // shouldProcessIngressUpdate checks whether we should renotify registered handlers about an update event
-func (c *controller) shouldProcessIngressUpdate(oldObj, curObj interface{}) (bool, error) {
-	var shouldProcess bool
-
+func (c *controller) shouldProcessIngressUpdate(curObj interface{}, event model.Event) (bool, error) {
 	// should always have curObj passed
 	ing, ok := curObj.(*knetworking.Ingress)
 	if !ok {
 		return false, nil
 	}
 
-	if oldObj == nil { // corresponds to additions and deletions of ingresses, update handlers if the current version should be targeted
-		shouldProcessUpdate, err := c.shouldProcessIngress(c.meshWatcher.Mesh(), ing)
-		if err != nil {
-			return false, err
-		}
-		shouldProcess = shouldProcessUpdate
-	} else { // this case corresponds to an update to an existing ingress resource
-		oldIng, ok := oldObj.(*knetworking.Ingress)
-		if !ok {
-			return false, nil
-		}
-
-		shouldProcessOld, err := c.shouldProcessIngress(c.meshWatcher.Mesh(), oldIng)
-		if err != nil {
-			return false, err
-		}
-		shouldProcessNew, err := c.shouldProcessIngress(c.meshWatcher.Mesh(), ing)
-		if err != nil {
-			return false, err
-		}
+	c.mutex.Lock()
+	_, preProcessed := c.ingresses[ing.Namespace+"/"+ing.Name]
+	if event == model.EventDelete {
```

Comment:
I think ideally we don't event have an `event` input. This is how k8s queues work - they just get the object key (name+namespace).